swagger.yaml

```diff
@@ -0,0 +1,333 @@
+swagger: "2.0"
+info:
+  description: "An API used to customise a dataset. Once the job has been created it can then be updated with filters to apply to the dataset. Once the job is ready it is expected that the client will update the job state to start the processing. Once processing has been started the job will be placed in a locked condition as no amendments can be done."
+  version: "1.0.0"
+  title: "Filter API"
+  license:
+    name: "Open Government Licence v3.0"
+    url: "http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
+basePath: "/v1"
+tags:
+- name: "Filter"
+  description: "Used to filter published datasets"
+schemes:
+- "http"
+parameters:
+  filter_id:
+    name: filter_id
+    type: string
+    required: true
+    description: "The unique  filter ID for customising a dataset"
+    in: path
+  dimension_name:
+    name: dimension_name
+    type: string
+    required: true
+    description: "The name of a dimension"
+    in: path
+  value:
+    name: value
+    type: string
+    required: true
+    description: "The a single value for a dimension"
+    in: path
+  dimension:
+    name: dimension
+    schema:
+      $ref: '#/definitions/Dimension'
+    required: true
+    description: "A dimension to filter the dataset"
+    in: body
+  filter_job:
+    name: filter_job
+    schema:
+      $ref: '#/definitions/NewJob'
+    required: true
+    description: "Model of all editable properties within a filter job"
+    in: body
+paths:
+  /filters:
+    post:
+      tags:
+      - "Filter"
+      summary: "Create a filter job for a dataset"
+      description: "Create a job so that dimensions can be added to filter a dataset"
+      produces:
+      - "application/json"
+      parameters:
+      - $ref: '#/parameters/filter_job'
+      responses:
+        200:
+          description: "Job was created"
+          schema:
+            $ref: '#/definitions/JobState'
+        500:
+          $ref: '#/responses/InternalError'
+  /filters/{filter_id}:
+    parameters:
+      - $ref: '#/parameters/filter_id'
+    get:
+      tags:
+      - "Filter"
+      summary: "Get a description of a created job"
+      description: "A jobId is required to access a customised dataset job. To create a jobId see the `Dataset API`"
```

Comment:
This description needs updating. Still references jobId and mentions creating a job via the dataset API.